fact_extractor/plugins/unpacking/generic_carver/code/generic_carver.py

```diff
@@ -22,7 +22,7 @@ def unpack_function(file_path, tmp_dir):
     '''
 
     logging.debug('File Type unknown: execute binwalk on {}'.format(file_path))
-    output = execute_shell_command(f'binwalk --extract --carve --signature --directory  {tmp_dir} {file_path}')
+    output = execute_shell_command(f'timeout 580 binwalk --extract --carve --signature --directory  {tmp_dir} {file_path}')
```

Comment:
what about turning this a config parameter?

Reply:
Hey! I'm agreeing with @dgutson here.

For transparency, it is probably for the best to convert this into a config param. @Oren-i, do you have capacities to address this change or shall our team allocate resources for this change internally? (keep in mind that it may take some time then)

Reply:
Hi @rhelmke, where should the configuration option should be located? I can do it if it is only a change to fact_extractor. If a change to fact_core is required I think it is best if your team do it, as it needs more coordination.

Reply:
How did you get to the conclusion that the timeout for the docker image is 600? The call in FACT (`FACT_core/src/unpacker/unpack_base.py:28`) doesn't set a timeout and the default value (in `FACT_core/src/helperFunctions/docker.py:41`) is 300 seconds. So maybe it would make sense to increase the timeout in FACT.
What is more, the "generic carver" is also called as a fallback if another unpacking method fails so it could make sense to increase the time buffer a bit.

As far as placement for the configuration is concerned, there currently is no place for putting configuration of unpacking plugins. In "FACT_core" you can add configuration for each plugin in `main.cfg` but the development of FACT_core and the extractor diverged quite a bit since the latter moved to its own project. We could rework how extraction plugins are structured and use a similar class-based approach as in FACT_core but this would definitely be out-of-scope for this PR. Therefore, IMHO it would probably be best to set it as a constant inside the plugin (`fact_extractor/plugins/unpacking/generic_carver/code/generic_carver.py`).

Reply:
Thanks @jstucke, I will look at that fix. You are probably correct, I got the 600 seconds value from how long FACT ran in my test before showing the exception, the difference was probably due to uncompressing other files before the one that causes the error. 

I changed the timeout to 280, and added the constant to the plugin.

Reply:
Converting to draft as the changes are going do cause unneeded timeouts when using fact_extractor in standalone mode. I think the correct way to implement this should be by getting the timeout value from fact_core as a parameter.

Reply:
> Thanks @jstucke, I will look at that fix.

After the merge of #95 the docker image of the extractor was rebuilt and pushed to DockerHub so you could simply pull the updated image to try it out. I would be very much interested in knowing how well that change in itself alleviates your timeout problems.